collections/schema.go

```diff
@@ -12,15 +13,36 @@ import (
 // methods for importing/exporting genesis data and for schema reflection for
 // clients.
 type Schema struct {
-	storeKey            storetypes.StoreKey
+	storeAccessor       func(context.Context) store.KVStore
 	collectionsByPrefix map[string]collection
 	collectionsByName   map[string]collection
 }
 
-// NewSchema creates a new schema from the provided store key.
-func NewSchema(storeKey storetypes.StoreKey) Schema {
+// NewSchema creates a new schema for the provided KVStoreService.
+func NewSchema(service store.KVStoreService) Schema {
+	return NewSchemaFromAccessor(func(ctx context.Context) store.KVStore {
+		return service.OpenKVStore(ctx)
+	})
+}
+
+// NewMemoryStoreSchema creates a new schema for the provided MemoryStoreService.
+func NewMemoryStoreSchema(service store.MemoryStoreService) Schema {
+	return NewSchemaFromAccessor(func(ctx context.Context) store.KVStore {
+		return service.OpenMemoryStore(ctx)
+	})
+}
+
+// NewSchemaFromAccessor creates a new schema for the provided store accessor
+// function. Modules built against versions of the SDK which do not support
+// the cosmossdk.io/core/appmodule APIs should use this method.
+// Ex:
+
+//	NewSchemaFromAccessor(func(ctx context.Context) store.KVStore {
+//			return sdk.UnwrapSDKContext(ctx).KVStore(kvStoreKey)
```

Comment:
Would it make sense to export a function in `sdk/types` like this to facilitate adoption?

```golang
func SdkContextSchemaAccessor(storeKey store.StoreKey) func(ctx context.Context) store.KVStore {
	return func(ctx context.Context) store.KVStore {
		return sdk.UnwrapSDKContext(ctx).KVStore(storeKey)
	}
}
```

Reply:
Sure, that could be added to `types` if it's useful. Either way the function is pretty small